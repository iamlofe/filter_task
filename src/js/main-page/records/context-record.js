import { Record } from 'immutable';

const CurrentContext = new Record({
    searchTitle: null,
    searchType: null,
    demesions: null,
    id: null,

});

export default class CurrentContextRecord extends CurrentContext {
    static parse(info) {
        return new CurrentContextRecord({
            title: info.title,
            id: info.id,
            searchTitle: info.searchTitle,
            searchType: info.searchType
        });
    }
}
