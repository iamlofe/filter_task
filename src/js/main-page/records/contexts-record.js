import { Record, List } from 'immutable';

import RecordDemisions from './dimensions-record';

const Context = new Record({
    id: null,
    title: null,
    listsDimensions: new List(),
    isAccess: true,
    isActive: false,
});

export default class RecordContext extends Context {
    static parse(context) {
        return new RecordContext({
            id: context.id,
            title: context.title,
            listsDimensions:
                new List(context.listsDimensions.map(dimension => RecordDemisions.parse(dimension, context.id))),
        });
    }
}
