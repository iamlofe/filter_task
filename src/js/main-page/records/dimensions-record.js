import {
    Record,
    List,
} from 'immutable';

import RecordResults from './results-record';

const Demisions = new Record({
    id: null,
    title: null,
    listsResults: new List(),
    isActive: false,
    isAccess: false,
    demisionsSave: []
})

export default class RecordDemisions extends Demisions {
    static parse(demisions, contextId) {
        return new RecordDemisions({
            id: demisions.id,
            title: demisions.title,
            contextId: contextId,
            listsResults: new List(demisions.listResults.map(r => RecordResults.parse(r, contextId, demisions.id))),
        })
    }
}