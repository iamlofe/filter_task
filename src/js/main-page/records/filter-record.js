import { Record } from 'immutable';

export const CurrentContext = new Record({
    searchTitle: null,
    searchType: null,
    isSaving: null,
    isRestoring: null,
    contextIds: null
});
