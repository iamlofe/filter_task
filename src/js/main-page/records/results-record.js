import {
    Record,
    List,
} from 'immutable';

import idGeneration from '../tools/generation-id-tool';

const Results = new Record({
    id: null,
    title: null,
    demisionId: null,
    contextId: null,
    isActive: false,
    isAccess: false,
    resultsSave: []
})


export default class RecordResults extends Results {
    static parse(result, ctxId, dimId) {
        return new RecordResults({
            id: result.id,
            title: result.title,
            demisionId: dimId,
            contextId: ctxId
        })
    }
}