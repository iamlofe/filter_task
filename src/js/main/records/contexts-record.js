import { Record, List } from 'immutable';

import RecordDemisions from './dimensions-record';

export const Context = new Record({
    id: null,
    contextId: null,
    title: null,
    listsDimensions: new List()
});

export default class ContextRecord extends Context {
    static parse(context) {
        return new ContextRecord({
            id: context.id,
            contextId: context.id,
            title: context.title,
            listsDimensions: new List(context.listsDimensions.map(dimension => RecordDemisions.parse(dimension, context.id)))
        });
    }
}
