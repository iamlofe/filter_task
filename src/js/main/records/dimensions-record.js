import { Record, List } from 'immutable';

import RecordResults from './results-record';

const Demisions = new Record({
    id: null,
    title: null,
    contextId: null,
    listsResults: new List(),
    demisionId: null
});

export default class DemisionsRecord extends Demisions {
    static parse(demisions, contextId) {
        return new DemisionsRecord({
            demisionId: demisions.id,
            title: demisions.title,
            contextId,
            listsResults: new List(demisions.listResults.map(r =>
                RecordResults.parse(r, contextId, demisions.id)))
        });
    }
}
