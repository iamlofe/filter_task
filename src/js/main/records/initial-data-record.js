import { Record, List } from 'immutable';

export default new Record({
    initialDataFilter: new List(),
    isStatusLoadData: true
});
