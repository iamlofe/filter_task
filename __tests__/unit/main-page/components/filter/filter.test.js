import React from 'react';
import { shallow } from 'enzyme';
import { List } from 'immutable';

import { CurrentContext } from 'main-page/records/filter-record';
import Filter from 'main-page/components/filter/filter';

describe('account component', () => {
    const baseProps = {
        filterId: '',
        isRestoreAvailable: false,
        onRestoreState: jest.fn(),
        onToggleWidget: jest.fn(),
        isWidgetOpen: false,
        onSaveDataWidget: jest.fn(),
        isSavingData: false,
        selectedContext: new List(),
        selectedDemision: new List(),
        selectedResults: new List(),
        contexts: new List(),
        demisions: new List(),
        results: new List(),
        onDeleteFilter: jest.fn(),
        isRestoringData: false,
        dataWidget: new CurrentContext()
    };

    it('is shallow rendered without crashing', () => {
        const filter = <Filter {...baseProps} />;

        const filterComponent = shallow(filter);
        expect(filterComponent).toMatchSnapshot();
    });

    it('is shallow rendered with isWidgetOpen true without crashing', () => {
        const filter = <Filter {...baseProps} isWidgetOpen />;

        const filterComponent = shallow(filter);
        expect(filterComponent.find('.filter-container__button_open-filter').prop('label')).toEqual('Close');
        expect(filterComponent).toMatchSnapshot();
    });

    it('is shallow rendered with isRestoreAvailable true without crashing', () => {
        const filter = <Filter {...baseProps} isRestoreAvailable />;

        const filterComponent = shallow(filter);
        expect(filterComponent).toMatchSnapshot();
    });

    it('is shallow rendered with isSavingData true without crashing', () => {
        const filter = <Filter {...baseProps} isSavingData />;

        const filterComponent = shallow(filter);
        expect(filterComponent.find('.filter-container__button_save-filter').prop('label')).toEqual('...');
        expect(filterComponent).toMatchSnapshot();
    });

    it('is shallow rendered with isRestoringData true without crashing', () => {
        const filter = <Filter {...baseProps} isRestoringData isRestoreAvailable />;

        const filterComponent = shallow(filter);
        expect(filterComponent.find('.filter-container__button_restore-filter').prop('label')).toEqual('...');
        expect(filterComponent).toMatchSnapshot();
    });

    it('is shallow rendered with dataWidget.searchTitle  without ', () => {
        const filter = (
            <Filter
                {...baseProps}
                dataWidget={
                    new CurrentContext({
                        searchTitle: 'test'
                    })
                }
            />
        );

        const filterComponent = shallow(filter);
        expect(filterComponent).toMatchSnapshot();
    });
});
