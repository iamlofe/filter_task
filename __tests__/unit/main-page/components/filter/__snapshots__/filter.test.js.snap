// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account component is shallow rendered with dataWidget.searchTitle  without  1`] = `
<div
  className="filter-container__state"
>
  <div
    className="filter-container__state-content"
  >
    <div
      className="filter-container__space-buttons"
    >
      <FilterButtonOpenWidget
        className="filter-container__button_open-filter"
        label="Open"
        onClick={[MockFunction]}
      />
      <FilterButtonOpenWidget
        className="filter-container__button_delete-filter"
        label="Delete"
        onClick={[MockFunction]}
      />
      <FilterButtonOpenWidget
        className="filter-container__button_save-filter"
        label="Save"
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="filter-container__display-state-container"
    >
      <div
        className="filter-container__display-state"
      >
        <div
          className="filter-container__parametrs-search"
        >
          <div
            className="filter-container__type-search"
          >
            Type: 
            <span />
          </div>
          <div
            className="filter-container__string-search"
          >
            Search: 
            <span>
              test
            </span>
          </div>
        </div>
        <FilterState
          label="Context"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
        <FilterState
          label="Demisions"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
        <FilterState
          label="Results"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`account component is shallow rendered with isRestoreAvailable true without crashing 1`] = `
<div
  className="filter-container__state"
>
  <div
    className="filter-container__state-content"
  >
    <div
      className="filter-container__space-buttons"
    >
      <FilterButtonOpenWidget
        className="filter-container__button_open-filter"
        label="Open"
        onClick={[MockFunction]}
      />
      <FilterButtonOpenWidget
        className="filter-container__button_delete-filter"
        label="Delete"
        onClick={[MockFunction]}
      />
      <FilterButtonOpenWidget
        className="filter-container__button_save-filter"
        label="Save"
        onClick={[MockFunction]}
      />
      <FilterButtonOpenWidget
        className="filter-container__button_restore-filter"
        label="Restore"
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="filter-container__display-state-container"
    >
      <div
        className="filter-container__display-state"
      >
        <div
          className="filter-container__parametrs-search"
        />
        <FilterState
          label="Context"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
        <FilterState
          label="Demisions"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
        <FilterState
          label="Results"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`account component is shallow rendered with isRestoringData true without crashing 1`] = `
<div
  className="filter-container__state"
>
  <div
    className="filter-container__state-content"
  >
    <div
      className="filter-container__space-buttons"
    >
      <FilterButtonOpenWidget
        className="filter-container__button_open-filter"
        label="Open"
        onClick={[MockFunction]}
      />
      <FilterButtonOpenWidget
        className="filter-container__button_delete-filter"
        label="Delete"
        onClick={[MockFunction]}
      />
      <FilterButtonOpenWidget
        className="filter-container__button_save-filter"
        label="Save"
        onClick={[MockFunction]}
      />
      <FilterButtonOpenWidget
        className="filter-container__button_restore-filter"
        label="..."
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="filter-container__display-state-container"
    >
      <div
        className="filter-container__display-state"
      >
        <div
          className="filter-container__parametrs-search"
        />
        <FilterState
          label="Context"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
        <FilterState
          label="Demisions"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
        <FilterState
          label="Results"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`account component is shallow rendered with isSavingData true without crashing 1`] = `
<div
  className="filter-container__state"
>
  <div
    className="filter-container__state-content"
  >
    <div
      className="filter-container__space-buttons"
    >
      <FilterButtonOpenWidget
        className="filter-container__button_open-filter"
        label="Open"
        onClick={[MockFunction]}
      />
      <FilterButtonOpenWidget
        className="filter-container__button_delete-filter"
        label="Delete"
        onClick={[MockFunction]}
      />
      <FilterButtonOpenWidget
        className="filter-container__button_save-filter"
        label="..."
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="filter-container__display-state-container"
    >
      <div
        className="filter-container__display-state"
      >
        <div
          className="filter-container__parametrs-search"
        />
        <FilterState
          label="Context"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
        <FilterState
          label="Demisions"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
        <FilterState
          label="Results"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`account component is shallow rendered with isWidgetOpen true without crashing 1`] = `
<div
  className="filter-container__state"
>
  <div
    className="filter-container__state-content"
  >
    <div
      className="filter-container__space-buttons"
    >
      <FilterButtonOpenWidget
        className="filter-container__button_open-filter"
        label="Close"
        onClick={[MockFunction]}
      />
      <FilterButtonOpenWidget
        className="filter-container__button_delete-filter"
        label="Delete"
        onClick={[MockFunction]}
      />
      <FilterButtonOpenWidget
        className="filter-container__button_save-filter"
        label="Save"
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="filter-container__display-state-container"
    >
      <div
        className="filter-container__display-state"
      >
        <div
          className="filter-container__parametrs-search"
        />
        <FilterState
          label="Context"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
        <FilterState
          label="Demisions"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
        <FilterState
          label="Results"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
      </div>
      <Portal>
        <Connect(FilterWidget)
          filterId=""
          onToggleWidget={[MockFunction]}
        />
      </Portal>
    </div>
  </div>
</div>
`;

exports[`account component is shallow rendered without crashing 1`] = `
<div
  className="filter-container__state"
>
  <div
    className="filter-container__state-content"
  >
    <div
      className="filter-container__space-buttons"
    >
      <FilterButtonOpenWidget
        className="filter-container__button_open-filter"
        label="Open"
        onClick={[MockFunction]}
      />
      <FilterButtonOpenWidget
        className="filter-container__button_delete-filter"
        label="Delete"
        onClick={[MockFunction]}
      />
      <FilterButtonOpenWidget
        className="filter-container__button_save-filter"
        label="Save"
        onClick={[MockFunction]}
      />
    </div>
    <div
      className="filter-container__display-state-container"
    >
      <div
        className="filter-container__display-state"
      >
        <div
          className="filter-container__parametrs-search"
        />
        <FilterState
          label="Context"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
        <FilterState
          label="Demisions"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
        <FilterState
          label="Results"
          listItems={Immutable.List []}
          selectedItems={Immutable.List []}
        />
      </div>
    </div>
  </div>
</div>
`;
